src/Adapter/CartRule/QueryHandler/GetCartRuleForEditingHandler.php

```diff
@@ -52,8 +54,13 @@
 /**
  * Handles command which gets catalog price rule for editing using legacy object model
  */
-final class GetCartRuleForEditingHandler extends AbstractCartRuleHandler implements GetCartRuleForEditingHandlerInterface
+class GetCartRuleForEditingHandler extends AbstractCartRuleHandler implements GetCartRuleForEditingHandlerInterface
```

Comment:
```suggestion
class GetCartRuleForEditingHandler implements GetCartRuleForEditingHandlerInterface
```
You shouldn't rely on this abstract class anymore as it uses legacy code and not the repositories. It should even be removed as it encourages relying on old code. Since you now inject the repository in this handler it shouldn't need the abstract class anymore

If you don't want to refacto all the current usages of the abstract it's fine, it can be done piece by piece or in a dedicated PR but at least here you can removed it, it's part of this PR's scope after all